### PR TITLE
Do not Assume all Fields are Provided when a Trigger is Posted

### PR DESCRIPTION
**Pull request by dubeejw**
- Ensure all needed fields are received from the client when posting a trigger
- Return an error JSON payload if proper fields are not received
- Don't error in Consumer if isMessageHub is set to false and credentials are not providedi

### CODE REVIEW

**Comment by bjustin-ibm** [comment]
@dubeejw I added a commit here (22045bcf56dde3ccb9602989a8a1dbcf7faf61e9) that simplifies this a little bit. Please review.

File: provider/app.py [hunk 1]
```diff
@@ -152,16 +152,16 @@ def restoreTriggers():
 
 def getMissingPostFields(fields):
     missing = []
-    messageHubRequired = ['brokers', 'topic', 'triggerURL', 'isMessageHub', 'username', 'password']
-    kafkaRequired = ['brokers', 'topic', 'triggerURL', 'isMessageHub']
+    requiredFields = ['brokers', 'topic', 'triggerURL', 'isMessageHub']
+
+    # Message Hub also requires 'username' and 'password fields'
+    if 'isMessageHub' in fields and fields['isMessageHub'] == True:
```
Comment (dubeejw) [inline provider/app.py]: What if fields is None here, does an exception occur?

Comment (bjustin-ibm) [inline provider/app.py]: Good point. Fixed in 0f85b6600cf6be8488c80b055a2ee56579d2cef9.

File: provider/app.py [hunk 2]
```diff
@@ -154,13 +154,13 @@ def getMissingPostFields(fields):
...
 
-    # Message Hub also requires 'username' and 'password fields'
-    if 'isMessageHub' in fields and fields['isMessageHub'] == True:
-        requiredFields.extend(['username', 'password'])
-
     if fields is None:
         missing.extend(requiredFields)
     else:
+        # Message Hub also requires 'username' and 'password fields'
+        if 'isMessageHub' in fields and fields['isMessageHub'] == True:
```
Comment (dubeejw) [inline provider/app.py]: LGTM

**Pull request merged**